datafusion/expr/src/expr.rs

```diff
@@ -1413,12 +1413,16 @@ impl Expr {
             .unwrap()
     }
 
+    /// Returns true if the expression node is volatile, i.e. whether it can return
+    /// different results when evaluated multiple times with the same input.
+    pub fn is_volatile_node(&self) -> bool {
```

Comment:
I think it would help to clarify in the docs the difference between `is_volatile` and `is_volatile_node`

Perhaps something like

```suggestion
    /// 
    /// Note: unlike [`Self::is_volatile`], this function does not consider inputs:
    /// - `rand()` returns `true`, 
    /// - `a + rand()` returns `false`
    pub fn is_volatile_node(&self) -> bool {
```

(and something similar for `is_volatile`)

Reply:
Added in https://github.com/apache/datafusion/commit/79b2e022678177358d8325996354ea10a34582fb.